### PR TITLE
Allow the user to configure more logging to be stored by xDrip.

### PR DESCRIPTION
**Pull request by tzachi-dar**
This is needed if the user has a problem in some component and we want to allow him to send more logs without creating a private build for him.

Once approved I'll remove the logs with tag "dddd" that were used to debug this code.

Tested by logging different log components.

### CODE REVIEW

File: app/src/main/java/com/eveningoutpost/dexdrip/Models/UserError.java [hunk 1]
```diff
@@ -151,52 +155,139 @@ public static void e(String a, String b){
             new UserError(a, b);
         }
 
-        public static void e(String a, String b, Exception e){
-            android.util.Log.e(a, b, e);
-            new UserError(a, b + "\n" + e.toString());
+        public static void e(String tag, String b, Exception e){
+            android.util.Log.e(tag, b, e);
+            new UserError(tag, b + "\n" + e.toString());
```
Comment (jamorham) [inline app/src/main/java/com/eveningoutpost/dexdrip/Models/UserError.java]: This and similar lines, is this change from `a` to `tag` just a refactor to make it more readable?

File: app/src/main/java/com/eveningoutpost/dexdrip/Models/UserError.java [hunk 2]
```diff
@@ -151,52 +155,139 @@ public static void e(String a, String b){
...
+    	}
+    	
+    	/*
+    	 * This function reads a string representing tags that the user wants to log
+    	 * Format of string is tag1:level1,tag2,level2
+    	 * Example of string is Alerts:i,BG:W
+    	 * 
+    	 */
+    	public static void readPreference(String extraLogs) {
+    		Log.e("dddd", "called " +  extraLogs);
```
Comment (jamorham) [inline app/src/main/java/com/eveningoutpost/dexdrip/Models/UserError.java]: This and similar lines, is `dddd` just an internal tag for your use when debugging this code or does it have any special significance?

**Comment by tzachi-dar** [comment]
OK,
I have moved things down in the menu to below the "Store Logs for Troubleshooting.
So I have removed all the dddd logging tags.

I have fixed the preferences conflicts.
I'm opening this as a new PR in order to make it more clear of what I have made.

**Comment by jamorham** [comment]
I'm pretty sure this was already merged from another PR so I am closing those one as I don't think it is required any more.

**Pull request closed without merging**